### PR TITLE
Suggested names for passport

### PR DESCRIPTION
**Pull request by bmallred**
Issue #404 

## TODO

 - [x] Get approval from @Jay-Neighbours-I72 on way prototype and way ahead
 - [x] Refactor in to reusable component (for example addresses)
 - [x] Update testing around suggestions

## Preview

passport-suggestions

### CODE REVIEW

**Comment by Jay-Neighbours-I72** [comment]
The address verification style is part of this sprint. I think we should wait to have an approved style on that because that style will be used here too.

**Comment by Jay-Neighbours-I72** [comment]
Address verification style was approved so good to go on this.

File: src/components/Form/Name/Name.jsx [hunk 1]
```diff
@@ -214,8 +216,98 @@ export default class Name extends ValidationElement {
...
   }
 
+  /**
+   * Return the possible suggestions or an empty value if there is nothing to present.
+   *
+   *  - Assign the property `withSuggestions` to turn this feature on
+   */
+  suggestions () {
+    // If suggestions have already been dismissed then skip
+    if (this.state.dismissSuggestions) {
```
Comment (adelevie) [inline src/components/Form/Name/Name.jsx]: These looks like great candidates for some named functions :)

**Comment by bmallred** [comment]
## Updated with modal

alternate_names

**NOTE**: Design guidelines are present in the screenshot

File: src/components/Form/Suggestions/Suggestions.test.jsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      withSuggestions: 'true',
+      suggestionTitle: 'Suggestion title',
+      suggestionParagraph: 'Suggestion paragraph',
+      suggestionLabel: 'Suggestion label',
+      suggestionDismissLabel: 'Dismiss',
+      suggestionUseLabel: 'Use'
+    }
+    const component = mount(<Suggestions {...expected}><span>Something</span></Suggestions>)
+    expect(component.find('.modal').length).toEqual(1)
+    component.find('.suggestion .action button').first().simulate('click')
```
Comment (floresj) [inline src/components/Form/Suggestions/Suggestions.test.jsx]: Nice, didn't know you could do more complex selectors.

**Pull request merged**